File: x_f259(frog)/x_Python/frog.py
from sys import stdin, stdout
def getstr():
    str = ''
    while True:
        ch = stdin.read(1)
        if ch == '\n' or not ch:
            return str
        str += ch

prev_length = 0
curr_length = 0
names = set()
while True:
    try:
        name = getstr()
        if not name: break
        names.add(name)
        curr_length = len(names)
        if curr_length != prev_length:
            stdout.write('0\n')
        else:
            stdout.write('1\n')
        prev_length = curr_length

File: f259(frog)/C++/frog.cpp
#include <iostream>
#include <set>
int main(void) {
  std::ios::sync_with_stdio(false);
  std::cin.tie(0);
  std::string name;
  std::set<std::string> names;
  int prev_length = 0, curr_length = 0;
  while (std::getline(std::cin, name)) {
    names.insert(name);
    curr_length = names.size();
    if (curr_length != prev_length) std::cout << "0\n";
    else std::cout << "1\n";
    prev_length = curr_length;
  }
  return 0;
}
